ssl/record/ssl3_buffer.c

```diff
@@ -117,18 +117,21 @@ int ssl3_setup_write_buffer(SSL *s, size_t numwpipes, size_t len)
         }
 
         if (thiswb->buf == NULL) {
-            p = OPENSSL_malloc(len);
-            if (p == NULL) {
-                s->rlayer.numwpipes = currpipe;
-                /*
-                 * We've got a malloc failure, and we're still initialising
-                 * buffers. We assume we're so doomed that we won't even be able
-                 * to send an alert.
-                 */
-                SSLfatal(s, SSL_AD_NO_ALERT,
-                         SSL_F_SSL3_SETUP_WRITE_BUFFER, ERR_R_MALLOC_FAILURE);
-                return 0;
-            }
+            if (!s->wbio || (s->wbio && !BIO_get_ktls(s->wbio))) {
```

Comment:
There is no need to test `s->wbio` twice. Also use `s->wbio == NULL`, i.e.
`if (s->wbio == NULL || !BIO_get_ktls(s->wbio)) {`

Reply:
Done.